Use a proper `MessageHandler` for `PartialEvaluator.getTextContent` to avoid errors for fonts relying on built-in CMap files (PR 8064 follow-up)

Snuffleupagus opened the pull request:
*My apologies for inadvertently breaking this in PR #8064; apparently we don't have any tests that cover this use-case :(*

Without this patch `getTextContent` will fail if called before `getOperatorList`, since loading of fonts during text-extraction may require fetching of built-in CMap files.

*Please note:* The `text` test added here, which uses an already existing PDF file, fails without this patch.

Given that this patch fixes a bad regression, I'm flagging a bunch of people for review (for whoever has time to look at it first).

---
Fixes #8193, courtesy of the second commit.

*Three cheers for all the idiosyncrasies in IE that made the second patch necessary :-P*

Snuffleupagus committed:
Use a proper `MessageHandler` for `PartialEvaluator.getTextContent` to avoid errors for fonts relying on built-in CMap files (PR 8064 follow-up)

*My apologies for inadvertently breaking this in PR 8064; apparently we don't have any tests that cover this use-case :(*

Without this patch `getTextContent` will fail if called before `getOperatorList`, since loading of fonts during text-extraction may require fetching of built-in CMap files.

*Please note:* The `text` test added here, which uses an already existing PDF file, fails without this patch.

Snuffleupagus commented:
/botio test

takahiroyoshi commented:
>This could potentially be a solution for issue #8193, but please note that I've not checked yet.
Provided that the way how I test this patch is right, this patch does not solve the issue #8193.

Snuffleupagus committed:
Ensure that the `XMLHttpRequest` is `open`ed before attempting to set the `responseType` in the `DOMCMapReaderFactory`, since IE fails otherwise (issue 8193)

I really cannot understand why this change is necessary, since modern browsers such as Firefox and Chrome work just fine with the old code.
Hence this is patch is yet another "hack" that's needed just because IE apparently cannot just work like you'd expect.

For consistency, the Node factory used in the CMap unit-tests is changed as well.

Fixes 8193.

yurydelendik commented:
Thank you for the patch.
